### ISSUE TITLE
Update translations

### ISSUE THREAD

**Issue by mariiaalt** [mpc-qt/mpc-qt#107]
Hello!
I would like to help updating Russian translation but I have some doubts about how I can do this. I found a Transifex project and tried to join the Russian project but failed — almost half a year my request is still unnoticed. Is it possible to somehow contribute to translations?

**Comment by cmdrkotori** [comment]
> I found a Transifex project and tried to join the Russian project but failed — almost half a year my request is still unnoticed

I lost access to the transifex project 5 years ago when my email address changed.  (Long story.)  I recently created a new project under my new email over 15 days ago but it has yet to be accepted or made public.  It should be showing up at https://explore.transifex.com/mpc-qt/.

>Is it possible to somehow contribute to translations?

I think we will have to go with another translation service.  Any suggestions?  It needs to be free because I barely have enough money to pay the bills.  I've been looking at hosted weblate fwiw.

**Comment by Shoun2137** [comment]
I don't know if you're keen on selfhosting, but what about https://weblate.org/ then?

**Comment by cmdrkotori** [comment]
Sounds good. I made a request for libre hosting on weblate.

**Comment by mariiaalt** [comment]
Weblate is a good option I think. As an alternative there is also https://crowdin.com/ but I am not sure if it is free or not. Or we (translators) can just make pull requests here in GitHub if it is comfortable for you.
Anyway, thanks for the information, I will be waiting for updates.

**Comment by cmdrkotori** [comment]
We're currently waiting on approval.

**Comment by mariiaalt** [comment]
Hello!
I found Weblate project and added there my translation but I see that here, in Github, my translation is missing. So I just want to check: does Weblate version work? Maybe it will be easier to update a translation and make pull request in Github?

**Comment by cmdrkotori** [comment]
Sorry, I have not been able to touch mpc-qt significantly since the end of May.

Yes, our weblate project got approved a few weeks ago, but I have yet to properly set up weblate.  I need to find time to devote to learning the webplate process and determining the best way to set it up.

I guess I can merge a PR in the mean time.  But I suppose in the future when weblate is fully set up you will not have to do that and can just use weblate.

I hope that clears it up for you.

**Comment by cmdrkotori** [comment]
Okay I managed to set up weblate we seem to have commits working (had to manually push the push button to test a few times in the last hour).  Time will tell if the repo get updated automatically.

**Comment by tbertels** [comment]
The newly added strings don't seem to get their English original translation added to Weblate, they just stay empty: https://hosted.weblate.org/translate/mpc-qt/mpc-qt-translations/en/?q=state:%3Ctranslated

**Comment by cmdrkotori** [comment]
> The newly added strings don't seem to get their English original translation added to Weblate, they just stay empty: https://hosted.weblate.org/translate/mpc-qt/mpc-qt-translations/en/?q=state:%3Ctranslated

Yeah the English "translation" needs a ton of copy-pasting when new strings are added if it's to show up as 100% translated.  I suspect it's a problem inherited from the way transifex wrote out the English translation file.  Anyway I fixed it, and those strings should show up here in 24 hours if weblate deems pushing necessary.

**Comment by tbertels** [comment]
Do you mean that every time a string gets added (or even modified?), you have to copy it manually on Weblate?

About Transifex, is the [.tx dir](https://github.com/mpc-qt/mpc-qt/tree/master/.tx) still needed?

**Comment by RaingodSpires** [comment]
I will help to translate to spanish and catalan.

**Comment by tbertels** [comment]
> > The newly added strings don't seem to get their English original translation added to Weblate, they just stay empty: https://hosted.weblate.org/translate/mpc-qt/mpc-qt-translations/en/?q=state:%3Ctranslated
> 
> Yeah the English "translation" needs a ton of copy-pasting when new strings are added if it's to show up as 100% translated. I suspect it's a problem inherited from the way transifex wrote out the English translation file. Anyway I fixed it, and those strings should show up here in 24 hours if weblate deems pushing necessary.

While fixing the merge conflict in translations/mpc-qt_zh_CN.ts, I've noticed that setting in Settings > Files:
"Monolingual base language file" which was set to `translations/mpc-qt_en.ts` instead of being kept empty.
So I've emptied it as we're using Qt Linguist in multilingual mode (both English and the translation in the same file).
Then I got an error about "Duplicated translation" because the English is both the source and a translation.
To fix that, I've just filtered English in the "Language filter" of that page by replacing `^[^.]+$` by `^(?!(en)$).+$`.

Let's hope we won't need to copy/paste the English translations anymore.

To fix the merge conflict, I've pulled the Weblate master branch (remote is https://hosted.weblate.org/git/mpc-qt/mpc-qt-translations/) in the master branch of this repository, fixed the merge conflict (linked to the Volume Up string) and pushed the updates to mpc-qt master.
Then on Weblate in Operations > Repository Maintenance for the mpc-qt-translations component, I've clicked on the red button "Reset and discard" which is described as

> Reset all changes in the Weblate repository

> Resets Weblate’s repository to match upstream and discards any pending changes in Weblate. Use when you want the upstream to be the authority for translations. It will discard changes in Weblate repo and bring it in sync with the upstream repository. 

Hopefully, this will make it easier to fix future merge conflicts.
